diff --git a/epan/CMakeLists.txt b/epan/CMakeLists.txt
index 34096c5..c6721de 100644
--- a/epan/CMakeLists.txt
+++ b/epan/CMakeLists.txt
@@ -581,6 +581,12 @@ set(DISSECTOR_SRC
 	dissectors/packet-giop.c
 	dissectors/packet-git.c
 	dissectors/packet-glbp.c
+	dissectors/packet-gluster_cli.c
+	dissectors/packet-gluster_dump.c
+	dissectors/packet-glusterfs.c
+	dissectors/packet-gluster_gd_mgmt.c
+	dissectors/packet-gluster_hndsk.c
+	dissectors/packet-gluster_pmap.c
 	dissectors/packet-gmhdr.c
 	dissectors/packet-gmrp.c
 	dissectors/packet-gnutella.c
diff --git a/epan/dissectors/Makefile.common b/epan/dissectors/Makefile.common
index c3d8018..671545c 100644
--- a/epan/dissectors/Makefile.common
+++ b/epan/dissectors/Makefile.common
@@ -494,6 +494,12 @@ DISSECTOR_SRC = \
 	packet-giop.c		\
 	packet-git.c		\
 	packet-glbp.c		\
+	packet-gluster_cli.c	\
+	packet-gluster_dump.c	\
+	packet-glusterfs.c	\
+	packet-gluster_gd_mgmt.c \
+	packet-gluster_hndsk.c	\
+	packet-gluster_pmap.c	\
 	packet-gmhdr.c		\
 	packet-gmrp.c		\
 	packet-gnutella.c	\
@@ -1133,6 +1139,7 @@ DISSECTOR_INCLUDES =	\
 	packet-frame.h	\
 	packet-ftam.h	\
 	packet-giop.h	\
+	packet-gluster.h	\
 	packet-gnm.h	\
 	packet-gnutella.h	\
 	packet-gprscdr.h \
